FTBFS with sshagent

<!--- Provide a general summary of the issue in the title above -->
When trying to build KeePassXC with the option `WITH_XC_SSHAGENT=ON` I get a compile error:
```
[13/143] Building CXX object src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o
FAILED: src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o 
/usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_EXCEPTIONS -DQT_STRICT_ITERATORS -DQT_WIDGETS_LIB -DWITH_APP_BUNDLE -I../src -Isrc -I../src/zxcvbn -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -pie -fPIE -std=c++11 -O2 -g -DNDEBUG   -fPIC -std=gnu++11 -MMD -MT src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o -MF src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o.d -o src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o -c ../src/sshagent/AgentSettingsPage.cpp
In file included from ../src/sshagent/AgentSettingsPage.cpp:20:0:
../src/sshagent/AgentSettingsWidget.h:24:36: fatal error: ui_AgentSettingsWidget.h: No such file or directory
compilation terminated.
[14/143] Building CXX object src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o
FAILED: src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o 
/usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_EXCEPTIONS -DQT_STRICT_ITERATORS -DQT_WIDGETS_LIB -DWITH_APP_BUNDLE -I../src -Isrc -I../src/zxcvbn -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -pie -fPIE -std=c++11 -O2 -g -DNDEBUG   -fPIC -std=gnu++11 -MMD -MT src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o -MF src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o.d -o src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o -c ../src/sshagent/AgentSettingsWidget.cpp
In file included from ../src/sshagent/AgentSettingsWidget.cpp:19:0:
../src/sshagent/AgentSettingsWidget.h:24:36: fatal error: ui_AgentSettingsWidget.h: No such file or directory
compilation terminated.
[15/143] Building CXX object src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o
FAILED: src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o 
/usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_EXCEPTIONS -DQT_STRICT_ITERATORS -DQT_WIDGETS_LIB -DWITH_APP_BUNDLE -I../src -Isrc -I../src/zxcvbn -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -pie -fPIE -std=c++11 -O2 -g -DNDEBUG   -fPIC -std=gnu++11 -MMD -MT src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o -MF src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o.d -o src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o -c src/sshagent/sshagent_automoc.cpp
In file included from src/sshagent/moc_AgentSettingsWidget.cpp:9:0,
                 from src/sshagent/sshagent_automoc.cpp:2:
src/sshagent/../../../src/sshagent/AgentSettingsWidget.h:24:36: fatal error: ui_AgentSettingsWidget.h: No such file or directory
```

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
No compile error

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from the current behavior -->
see above 

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->
Patching `src/sshagent/AgentSettingsWidget.h` helps:
Replace `#include "ui_AgentSettingsWidget.h"` with `#include "sshagent/ui_AgentSettingsWidget.h"`


## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Use KDE neon 
2. Use out of tree build
3. Build project (ninja/make is irrelevant)


Operating system: 
Description:    KDE neon User Edition 5.11
Release:        16.04
Codename:       xenial

CPU architecture: x64
Kernel: Linux nic-x250 4.10.0-40-generic
GCC: 5.4.0

This issue also breaks recent TeamCity builds. For example:
```
[07:32:18]In file included from /opt/buildagent/work/7ce463a40a422d79/src/sshagent/AgentSettingsWidget.cpp:19:
[07:32:18]/opt/buildagent/work/7ce463a40a422d79/src/sshagent/AgentSettingsWidget.h:24:10: fatal error: 'ui_AgentSettingsWidget.h' file not found
[07:32:18]#include "ui_AgentSettingsWidget.h"
[07:32:18]         ^
[07:32:18]In file included from /opt/buildagent/work/7ce463a40a422d79/build/src/sshagent/sshagent_automoc.cpp:2:
[07:32:18]In file included from /opt/buildagent/work/7ce463a40a422d79/build/src/sshagent/moc_AgentSettingsWidget.cpp:9:
[07:32:18]/opt/buildagent/work/7ce463a40a422d79/build/src/sshagent/../../../src/sshagent/AgentSettingsWidget.h:24:10: fatal error: 'ui_AgentSettingsWidget.h' file not found
[07:32:18]#include "ui_AgentSettingsWidget.h"
[07:32:18]         ^
[07:32:18]1 error generated.
[07:32:18]make[2]: *** [src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsWidget.cpp.o] Error 1
[07:32:18]make[2]: *** Waiting for unfinished jobs....
[07:32:18]In file included from /opt/buildagent/work/7ce463a40a422d79/src/sshagent/AgentSettingsPage.cpp:20:
[07:32:18]/opt/buildagent/work/7ce463a40a422d79/src/sshagent/AgentSettingsWidget.h:24:10: fatal error: 'ui_AgentSettingsWidget.h' file not found
[07:32:18]#include "ui_AgentSettingsWidget.h"
[07:32:18]         ^
[07:32:19]1 error generated.
[07:32:19]make[2]: *** [src/sshagent/CMakeFiles/sshagent.dir/AgentSettingsPage.cpp.o] Error 1
[07:32:19]1 error generated.
[07:32:19]make[2]: *** [src/sshagent/CMakeFiles/sshagent.dir/sshagent_automoc.cpp.o] Error 1
[07:32:20]make[1]: *** [src/sshagent/CMakeFiles/sshagent.dir/all] Error 2
[07:32:20]make[1]: *** Waiting for unfinished jobs....
[07:32:22]make: *** [all] Error 2
[07:32:23]Process exited with code 2
```
(From https://ci.keepassxc.org/viewLog.html?buildId=3348&tab=buildResultsDiv&buildTypeId=KeepassXC_DebugClang)

--- a/src/sshagent/AgentSettingsWidget.cpp
+++ b/src/sshagent/AgentSettingsWidget.cpp
@@ -17,6 +17,7 @@
  */
 
 #include "AgentSettingsWidget.h"
+#include "ui_AgentSettingsWidget.h"
 #include "core/Config.h"
 
 AgentSettingsWidget::AgentSettingsWidget(QWidget* parent)
@@ -24,6 +25,10 @@ AgentSettingsWidget::AgentSettingsWidget(QWidget* parent)
     , m_ui(new Ui::AgentSettingsWidget())
 {
     m_ui->setupUi(this);
+}
+
+AgentSettingsWidget::~AgentSettingsWidget()
+{
 }
 
 void AgentSettingsWidget::loadSettings()

--- a/src/sshagent/AgentSettingsWidget.h
+++ b/src/sshagent/AgentSettingsWidget.h
@@ -21,7 +21,6 @@
 
 #include <QWidget>
 #include <QScopedPointer>
-#include "ui_AgentSettingsWidget.h"
 
 namespace Ui {
     class AgentSettingsWidget;
@@ -32,8 +31,7 @@ class AgentSettingsWidget : public QWidget
     Q_OBJECT
 public:
     explicit AgentSettingsWidget(QWidget* parent = nullptr);
-
-signals:
+    ~AgentSettingsWidget();
 
 public slots:
     void loadSettings();

--- a/src/sshagent/CMakeLists.txt
+++ b/src/sshagent/CMakeLists.txt
@@ -1,4 +1,6 @@
 if(WITH_XC_SSHAGENT)
+    include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
+
     set(sshagent_SOURCES
         bcrypt_pbkdf.cpp
         blowfish.c

